tests/periph_pwm/main.c

```diff
@@ -20,42 +20,141 @@
  * every 1s on every channel.
  *
  * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Semjon Kerner <semjon.kerner@fu-berlin.de>
  *
  * @}
  */
 
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
 
 #include "xtimer.h"
+#include "irq.h"
+#include "shell.h"
 #include "timex.h"
 #include "periph/pwm.h"
 
-#define INTERVAL    (10LU * US_PER_MS) /* 10 ms */
-#define STEP        (10)
+#define OSC_INTERVAL    (10LU * US_PER_MS) /* 10 ms */
+#define OSC_STEP        (10)
+#define OSC_MODE        PWM_LEFT
+#define OSC_FREQU       (1000U)
+#define OSC_STEPS       (1000U)
+#define PWR_SLEEP       (1U)
 
-#define MODE        PWM_LEFT
-#define FREQU       (1000U)
-#define STEPS       (1000U)
+bool initiated;
```

Comment:
something does not quite fit: what happens if there a device supports multiple PWM devices? E.g.
```
init 0 ...
set 1 xx
```
This will work, as `initiated` is set to `true` while initializing PWM_DEV(0), but does not yield the wanted behavior...

So I would suggest making this variable a bitfield:
```c
static uint32_t initialized = 0;

....
// in init()
initialized |= (1 << dev);

// in set()
if (initialized & (1 << dev) {
    ...
}
```

Reply:
oh, and of course it needs to be declared `static` :-)